nexus/db-queries/src/db/datastore/update.rs

```diff
@@ -460,6 +438,58 @@ impl DataStore {
             .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))
     }
 
+    /// List all TUF repositories (without artifacts) ordered by system version (newest first by default).
+    pub async fn tuf_repo_list(
+        &self,
+        opctx: &OpContext,
+        pagparams: &DataPageParams<'_, Version>,
+    ) -> ListResultVec<TufRepo> {
+        opctx.authorize(authz::Action::Read, &authz::FLEET).await?;
+
+        use nexus_db_schema::schema::tuf_repo;
+
+        let conn = self.pool_connection_authorized(opctx).await?;
+
+        let marker_owner = pagparams
+            .marker
+            .map(|version| SemverVersion::from(version.clone()));
+        let db_pagparams = DataPageParams {
+            marker: marker_owner.as_ref(),
+            direction: pagparams.direction,
+            limit: pagparams.limit,
+        };
+
+        paginated(tuf_repo::table, tuf_repo::system_version, &db_pagparams)
+            .select(TufRepo::as_select())
+            .load_async(&*conn)
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))
+    }
+
+    /// List artifacts for a specific TUF repository by system version.
+    pub async fn tuf_repo_artifacts_list_by_version(
+        &self,
+        opctx: &OpContext,
+        system_version: SemverVersion,
+        _pagparams: &DataPageParams<'_, Uuid>,
+    ) -> ListResultVec<TufArtifact> {
+        opctx.authorize(authz::Action::Read, &authz::FLEET).await?;
+
+        let conn = self.pool_connection_authorized(opctx).await?;
+
+        // First get the repo by version
+        let repo = self.tuf_repo_get_by_version(opctx, system_version).await?;
+
+        // Get all artifacts for this repo and apply simple pagination
+        let all_artifacts = artifacts_for_repo(repo.id.into(), &conn)
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+
+        // For now, return all artifacts since each repo should only have a few (under 20)
+        // The existing artifacts_for_repo comment mentions this limitation
```

Comment:
I didn't think this was true any more.  I thought there was a version of #9109 where @iliana had paginated that precisely because these repos are much larger now, but I can't find that.  I'm okay with not paginating for an integration test but then I'd want to do the stuff I mentioned to make this harder to misuse.